glide-core/src/client/mod.rs

```diff
@@ -239,11 +239,32 @@ impl Client {
         run_with_timeout(request_timeout, async move {
             match self.internal_client {
                 ClientWrapper::Standalone(ref mut client) => client.send_command(cmd).await,
-
                 ClientWrapper::Cluster { ref mut client } => {
-                    let routing = routing
-                        .or_else(|| RoutingInfo::for_routable(cmd))
-                        .unwrap_or(RoutingInfo::SingleNode(SingleNodeRoutingInfo::Random));
+                    let routing =
+                        if let Some(RoutingInfo::SingleNode(SingleNodeRoutingInfo::Random)) =
+                            routing
+                        {
+                            let cmdname = cmd.command().unwrap_or_default();
+                            let cmdname = String::from_utf8_lossy(&cmdname);
+                            if redis::cluster_routing::is_readonly_cmd(cmdname.as_bytes()) {
+                                // A read-only command, go ahead and send it to a random node
+                                RoutingInfo::SingleNode(SingleNodeRoutingInfo::Random)
+                            } else {
+                                // A "Random" node was selected, but the command is a "@write" command
+                                // change the routing to "RandomPrimary"
+                                log_warn(
+                                    "send_command",
+                                    format!(
+                                        "User provided 'Random' routing which is not suitable for the writeable command '{cmdname}'. Changing it to 'RandomPrimary'"
+                                    ),
+                                );
+                                RoutingInfo::SingleNode(SingleNodeRoutingInfo::RandomPrimary)
+                            }
+                        } else {
+                            routing
```

Comment:
arent we getting back to the same place in case its not a RO command?

Reply:
The above code fixes a case (as reported by the OP) that the user might provides a "Random" routing but the command is not "Read-only". For other cases, we use the routing provided (and this is what we are doing on line `264`)